src/io/parquet/write.rs

```diff
@@ -50,23 +107,22 @@ impl ParquetWriterBuilder {
     }
 
     /// Consume the current builder to build a new writer.
-    pub fn build(self) -> Result<ParquetWriter> {
-        let writer = AsyncArrowWriter::try_new(
-            self.writer,
-            self.arrow_schema,
-            self.buffer_size,
-            self.props,
-        )?;
-
-        Ok(ParquetWriter { writer })
+    pub fn build(self) -> Result<(ParquetWriter, Arc<AtomicUsize>)> {
```

Comment:
I don't like the idea to return this in build. I prefer to change the API of `write()` to return the `(num_rows, written_size)`. But we can move on first to build our first table write PoC and polish here later.

Reply:
> Please note that the size is incorrect until we call close (data could be still in buffer).

Yes, AsyncArrowWriter will buffer the data. So when we call write(), it don't means that the data will be writed.
So for `write(&record_batch)->(num_rows,writtern_size)`, Dose the writtern_size means size of record_batch? or the writtern_size return by writer?

Reply:
> the writtern_size return by writer

We still need to wrap the writer, but don't need to expose to users.